### ISSUE TITLE
Executing a series of lines one after the other does not respect the execution order

### ISSUE THREAD

**Issue by dpsanders** [JunoLab/atom-julia-client#362]
In IJulia, executing a series of cells one after the other respects the order of execution. 
In Juno, apparently it doesn't.

**Comment by pfitzseb** [comment]
That's the first time I've heard of out-of-order execution problems in Juno. Can you reproduce with something like
```julia
global counter = 0

f() = (sleep(3); global counter += 1)

f()
f()
f()
f()
```

**Comment by dpsanders** [comment]
It seems to happen when recompiling packages for a long time and then trying to use things from those packages.

**Comment by pfitzseb** [comment]
Where "those packages" are packages from the Juno stack? I'll try to be on the lookout for this, but not sure how much I can to without being able to reproduce this.

**Comment by dpsanders** [comment]
No, sorry, my packages that need precompiling. Will make a mwe when I can.
